drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -92,18 +109,37 @@ static esp_err_t eth_esp32_rx(void *buffer, uint16_t len, void *eb)
 	return ESP_FAIL;
 }
 
-/* internally used by wifi hal layer */
-void esp_wifi_set_net_state(bool state)
+void esp_wifi_event_task(void)
 {
-	if (esp32_wifi_iface == NULL) {
-		LOG_ERR("network interface unavailable");
-		return;
-	}
-
-	if (state) {
-		net_if_up(esp32_wifi_iface);
-	} else {
-		net_if_down(esp32_wifi_iface);
+	int32_t event_id;
+
+	while (1) {
+		k_msgq_get(&esp_wifi_msgq, &event_id, K_FOREVER);
+
+		switch (event_id) {
+		case ESP32_WIFI_EVENT_STA_START:
```

Comment:
`ESP32_WIFI_EVENT_STA_START` is introduced in subsequent commit, right? Seems that it breaks bisectability then.

Reply:
@mniestroj I squash the commits in other to avoid that.